haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/GhcApiSupport.hs

```diff
@@ -229,6 +243,61 @@ withTypechecked globalLock GhcArgs{..} analysisOpts xrefSink
             -- This might be needed if TH executes code from other package? Or
             -- only if that code needs FFI?
 
+-- | Doc URI decls are for things that don't have source code available; most
+-- likely those from core packages. Hence their decls may be duplicated coming
+-- from multiple analyses. Deduplicate them, and put them in the 'xrefDocDecls'
+-- field of the first 'XRef'.
+attachDocUriDecls :: [XRef] -> [XRef]
+attachDocUriDecls graph =
+    let tickRefs = concatMap xrefCrossRefs graph
+        docDecls = generateDocUriDecls tickRefs
+     in case graph of
+            [] -> []
+            x : xs -> x {xrefDocDecls = docDecls} : xs
+
+generateDocUriDecls :: [TickReference] -> [DocUriDecl]
+generateDocUriDecls rs =
+  let declMap =
```

Comment:
Consider first deduping the `refTargetTicks` based on the `(PkgModule, name)` , and then generating the docUris on the deduped list. This should avoid lot of uri reconstruction.

Something like:

```
data Deduping = Deduping { dedupTick :: Tick, dedupKey :: (PkgModule, Text) }

let dedupedPotentialDocUriTargetTicks = map dedupTick
    . nubOrdOn dedupKey
    . map (\t -> Deduping t (tickPkgModule t, tickThing t))
    . filter (\t -> tickUniqueInModule t && isNothing (tickSpan t))
    . map refTargetTick
    $ refs
```

Reply:
That's better. Thanks!